src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -43,14 +59,71 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int process_resume(void) {
-        _cleanup_free_ char *name = NULL, *lnk = NULL;
+static int swap_contains_hibernation_image(DissectedPartition *p) {
+        _cleanup_close_ int fd = -1;
+
+        log_debug("Checking %s for hibernation signature", p->node);
+
+        fd = open(p->node, O_RDONLY|O_CLOEXEC);
+        if (fd < 0)
+                return log_error_errno(errno, "%s: cannot open: %m", p->node);
+
+        if (lseek(fd, arg_resume_offset * 512, SEEK_SET) < 0)
+                return log_error_errno(errno, "%s: cannot seek to position %"PRIu64": %m",
+                                       p->node, arg_resume_offset * 512);
+
+        uint8_t buf[10];
+        ssize_t len;
+
+        len = read(fd, buf, sizeof(buf));
```

Comment:
not that it matters but i think pread() is a bit sexier for cases like this, where we know we have a random access file...